### DISCUSSION TITLE
You should be able to right click on tablet without an external driver.

### DISCUSSION THREAD

**Discussion by dumbuz** [ppy/osu#25717] [Ideas]
Currently (to my knowledge) there is no way to pull up the context menu in song select or anywhere else with the built in OpenTabletDriver.

Holding down the pen would be a good option, but I think this also needs to be bindable to a pen button in the configuration.

**Comment by peppy** [comment]
Having an optional binding for right clicking sounds like something we definitely want.

**Reply by Marrasco** [reply]
Internal tablet driver feels great but can't use it since used to navigating song select with pen buttons. bump
(edit: having both external and internal drivers working sort of works, but for some reason left click does not hold down and only clicks while right can hold.)

**Reply by vatei** [reply]
Digging this up because as osu! becomes more and more reliant on context menus for important actions (which isn't a good or bad thing, just noticing), the current experience when only using a pen is still under what I would expect as a tablet user.
I know it's also possible to drop the pen and use the mouse, but I find that it tends to break my focus and flow, or I end up changing my grip accidentally. 
I think it feels much more natural to use the pen for everything 👍 

To account for this, I manually added a bind for right click on my keyboard, but it still feels clunky and frustrating to use unfortunately.
I'd suggest modifier + pen click, for example Ctrl+Pen click to bring up context menus, which feels like the most intuitive and comfortable to use with a pen in my opinion.
Thanks!

**Reply by Siiiuuuuu** [reply]
@vatei https://github.com/ppy/osu/discussions/32388 i had the exact same issue as you, here is what I proposed

**Reply by vatei** [reply]
Yeah but it's not just "use these mods", it's all context menus based on right click.
Holding pen tip to right click might also work but it sounds more tedious than ctrl+tip in my opinion.

**Reply by Siiiuuuuu** [reply]
ideally wed have something for both

**Reply by cerfeque** [reply]
we need this!

**Reply by darkralts** [reply]
> Having an optional binding for right clicking sounds like something we definitely want.

true mr peppy, perhaps have it to be the pen side button like most default wacom, i think most osu players are used to this.
